server/src/main/java/io/crate/expression/symbol/ScopedSymbol.java

```diff
@@ -54,11 +57,19 @@ public final class ScopedSymbol implements Symbol {
     private final RelationName relation;
     private final ColumnIdent column;
     private final DataType<?> dataType;
+    @Nullable
+    private final Symbol symbol;
 
-    public ScopedSymbol(RelationName relation, ColumnIdent column, DataType<?> dataType) {
+    public ScopedSymbol(RelationName relation, ColumnIdent column, DataType<?> dataType, @Nullable Symbol symbol) {
```

Comment:
Isn't it possible to track the association local in the union select? 

Adding a child symbol makes it much more difficult to reason about behaviour because of the symbol visiting functions across the code base. Will it traverse or not,  do the `any(ColumnIdent ...)` match on both and so on. 

Would be really great if we could avoid this. 

please let me have a closer look at the issue before going ahead with this

Reply:
OK, thank you

Reply:
Please have a look at the latest commit. Instead of making the scoped symbols distinguishable, Union's pruning logic was fixed.